edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Edgelet/versioning/ModuleManagementHttpClientVersioned.cs

```diff
@@ -92,8 +92,24 @@ public virtual async Task<Stream> GetModuleLogs(string module, bool follow, Opti
                 string baseUrl = HttpClientHelper.GetBaseUrl(this.ManagementUri).TrimEnd('/');
                 var logsUrl = new StringBuilder();
                 logsUrl.AppendFormat(CultureInfo.InvariantCulture, LogsUrlTemplate, baseUrl, module, this.Version.Name, follow.ToString().ToLowerInvariant());
-                since.ForEach(s => logsUrl.AppendFormat($"&{LogsUrlSinceParameter}={Uri.EscapeUriString(s)}"));
-                until.ForEach(u => logsUrl.AppendFormat($"&{LogsUrlUntilParameter}={Uri.EscapeUriString(u)}"));
+                since.ForEach(s =>
+                {
+                    if (!DateTime.TryParseExact(s, "yyyy-MM-dd'T'HH:mm:ssZ", CultureInfo.InvariantCulture, DateTimeStyles.None, out var time))
+                    {
+                        throw new ArgumentException($"Since Time with value : {s} is not in the correct format. The correct format should be yyyy-MM-dd'T'HH:mm:ssZ");
+                    }
+
+                    logsUrl.AppendFormat($"&{LogsUrlSinceParameter}={Uri.EscapeUriString(s)}");
+                });
+                until.ForEach(u =>
+                {
+                    if (!DateTime.TryParseExact(u, "yyyy-MM-dd'T'HH:mm:ssZ", CultureInfo.InvariantCulture, DateTimeStyles.None, out var time))
+                    {
+                        throw new ArgumentException($"Until Time with value : {u} is not in the correct format. The correct format should be yyyy-MM-dd'T'HH:mm:ssZ");
+                    }
```

Comment:
nit: There is probably a way to consolidate this dup logic. Quick thought is to put since and until into some collection then use some linq utility method. Massaging the message to say what you want is a bit tricker.

Reply:
@and-rewsmith : Done